inc/template-tags.php

```diff
@@ -23,6 +23,54 @@ function responsive_get_title() {
 	}
 }
 
+/**
+ * Displays the current page title.
+ *
+ * Functions like `the_title()`, but can be filtered to prevent or change the output
+ * intended for the main page title. Accepts the same arguments as `the_title()`, with
+ * an additional argument for $post->ID.
+ *
+ * Note: Intended only to be used where the main page title is.
+ *       Not intended to be used for things like a listing of posts.
+ *
+ * @since    2.1.4
+ *
+ * @link     https://codex.wordpress.org/Function_Reference/the_title
+ * @link     https://developer.wordpress.org/reference/hooks/the_title/
+ *
+ * @param    string $before Text to insert before the title, such as an opening h1 tag.
+ * @param    string $after  Text to insert after the title, such as a closing h1 tag.
+ * @param    bool   $echo   Prints the title to the screen when called. Defaults to true.
+ * @param    int    $id     Retrieves the title for a given post id.
+ */
+function responsive_the_title( $before = '', $after = '', $echo = true, $id = false ) {
+
+	/**
+	 * Filters the current page title.
+	 *
+	 * Useful for when something besides `get_the_title()` for the current query
+	 * should be used.
+	 *
+	 * @since    2.1.4
+	 */
+	$title = apply_filters( 'responsive_filter_page_title', get_the_title( $id ) );
+
+	// Only continues if a title exists and wasn't removed by the filter.
+	if ( ! empty( $title ) ) {
+
+		// Apply the normal `the_title` filters.
+		$title = apply_filters( 'the_title', $before . $title . $after );
```

Comment:
Can you please look into how this works a little further? I added a test `the_title` filter in functions.php and appended a string to the title. When calling `responsive_the_title()` it works fine. However, the filter seems to cause the `the_title` filter to be called more than once. The menus titles have two strings appended to them.

This might be a good clue to why the filters are being applied multiple times in `BU-Banners` as well.
This is the output of `responsive_the_title()`
![screen shot 2018-07-25 at 1 37 58 pm](https://user-images.githubusercontent.com/17765399/43217585-06d3bf2c-9010-11e8-831b-2210c2717bfc.png)

This is the what the main nav looks like in this test.
![screen shot 2018-07-25 at 1 37 52 pm](https://user-images.githubusercontent.com/17765399/43217586-06dd4cea-9010-11e8-8e78-6d7b08b4edfd.png)

Reply:
This is expected behavior for the nav menu items that are page titles to have the filter called twice. We can learn why this is expected behavior by looking through the source code for `get_the_title()` and the `Walker_Nav_Menu` class that generates WP nav menu markup.

First, look at the page title functions defined in `/wp-includes/post-template.php`.

- Line 43: The function `the_title()` is a wrapper function that actually calls `get_the_title()`
- Line 158: The function `get_the_title()` applies `the_title` filter whenever its called.

This means `the_title` filter is run anytime you call `get_the_title()` or `the_title()`. This is your first occurrence of the filter being applied.

Second, look at the `Walker_Nav_Menu` class defined in `/wp-includes/class-walker-nav-menu.php`.

- Line 204: You'll see that all nav menu item titles apply `the_title` filter again.

This is the second occurrence of the filter being applied; WordPress is again applying `the_title` filter to its nav menu titles. My understanding is that not every menu item is a page title, for example: a custom menu link. This is why WordPress applies `the_title` filter again to menu-item titles, since `get_the_title()` may not have been called for any custom written menu item titles.

As for BU Banners calling filters multiple times, that may be a separate issue that needs to be investigated.